contracts/socket/andromeda-socket-astroport/src/astroport.rs

```diff
@@ -0,0 +1,287 @@
+use andromeda_std::{
+    ado_contract::ADOContract,
+    amp::{
+        messages::{AMPMsg, AMPPkt},
+        AndrAddr, Recipient,
+    },
+    common::{context::ExecuteContext, denom::Asset},
+    error::ContractError,
+};
+use astroport::{
+    asset::AssetInfo,
+    router::{
+        Cw20HookMsg as AstroCw20HookMsg, ExecuteMsg as AstroExecuteMsg, QueryMsg as AstroQueryMsg,
+        SwapOperation as AstroSwapOperation,
+    },
+};
+use cosmwasm_std::{
+    attr, coin, ensure, to_json_binary, wasm_execute, Coin, Decimal, Deps, DepsMut, Env, Reply,
+    Response, StdError, SubMsg, Uint128, WasmMsg,
+};
+use cw20::{BalanceResponse, Cw20ExecuteMsg, Cw20QueryMsg};
+
+use crate::state::{ForwardReplyState, FORWARD_REPLY_STATE, PREV_BALANCE, SWAP_ROUTER};
+
+use andromeda_socket::astroport::{SimulateSwapOperationResponse, SwapOperation};
+
+pub const ASTROPORT_MSG_SWAP_ID: u64 = 1;
+pub const ASTROPORT_MSG_FORWARD_ID: u64 = 2;
+
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn execute_swap_astroport_msg(
+    ctx: ExecuteContext,
+    from_asset: Asset,
+    from_amount: Uint128,
+    to_asset: Asset,
+    recipient: Recipient,  // receiver where the swapped token goes to
+    refund_addr: AndrAddr, // refund address
+    max_spread: Option<Decimal>,
+    minimum_receive: Option<Uint128>,
+    operations: Option<Vec<SwapOperation>>,
+) -> Result<SubMsg, ContractError> {
+    let ExecuteContext { deps, env, .. } = ctx;
+
+    // Prepare offer and ask asset
+    ensure!(from_asset != to_asset, ContractError::DuplicateTokens {});
+    let from_denom = match from_asset.clone() {
+        Asset::NativeToken(denom) => denom,
+        Asset::Cw20Token(andr_addr) => andr_addr.get_raw_address(&deps.as_ref())?.to_string(),
+    };
+
+    // Prepare swap operations
+    let operations: Vec<AstroSwapOperation> = operations
+        .unwrap_or(vec![SwapOperation {
+            offer_asset_info: from_asset.clone(),
+            ask_asset_info: to_asset.clone(),
+        }])
+        .iter()
+        .map(|oper| {
+            let astro_operation = AstroSwapOperation::AstroSwap {
+                offer_asset_info: generate_asset_info_from_asset(
+                    &deps.as_ref(),
+                    oper.offer_asset_info.clone(),
+                )?,
+                ask_asset_info: generate_asset_info_from_asset(
+                    &deps.as_ref(),
+                    oper.ask_asset_info.clone(),
+                )?,
+            };
+            Ok(astro_operation)
+        })
+        .collect::<Result<Vec<AstroSwapOperation>, ContractError>>()?;
+    ensure!(
+        FORWARD_REPLY_STATE
+            .may_load(deps.as_ref().storage)?
+            .is_none(),
+        ContractError::Unauthorized {}
+    );
+
+    let amp_ctx = if let Some(pkt) = ctx.amp_ctx.clone() {
```

Comment:
Can be simplified into `let amp_ctx = ctx.amp_ctx.map(|pkt| pkt.ctx);`. 
There's also another occurrence of it